src/structures/ApplicationCommand.ts

```diff
@@ -0,0 +1,60 @@
+import BulbBotClient from "./BulbBotClient";
+import { CommandInteraction, Permissions, PermissionString } from "discord.js";
+import { ApplicationCommandOptions, ApplicationCommandType } from "../utils/types/ApplicationCommands";
+import { translateSlashCommands } from "../utils/InteractionCommands";
+
+export default class ApplicationCommand {
+	public readonly client: BulbBotClient;
+	public readonly type: ApplicationCommandType;
+	public readonly name: string;
+	public readonly description: string;
+	public readonly dm_permissions: boolean;
+	public readonly default_member_permissions: string | null;
+	public readonly command_permissions: PermissionString[];
+	public readonly options: ApplicationCommandOptions[];
+
+	constructor(client: BulbBotClient, { type, name, description, dm_permissions, command_permissions, options }: any) {
+		this.client = client;
+		this.type = type;
+		this.name = name;
+		this.description = description;
+		this.dm_permissions = dm_permissions || false;
+		this.command_permissions = command_permissions || [];
+		this.default_member_permissions = this._computePermissions();
+		this.options = this.appendTranslation(options) || [];
+	}
+
+	private applyTranslation(options: ApplicationCommandOptions[], name: string) {
+		for (let i = 0; i < options.length; i++) {
+			const optionCommand = options[i];
+			optionCommand.name_localizations = translateSlashCommands(`${name}_${optionCommand.name}_name`);
+			optionCommand.description_localizations = translateSlashCommands(`${name}_${optionCommand.name}_desc`);
+
+			if (`${name}_${optionCommand.name}_name`.length > 32 || `${name}_${optionCommand.name}_desc`.length > 32)
+				throw Error(`Too long of a name for slash commands: ${optionCommand.name} is too long for the slashcommand to register (over 32 characters)`);
+
+			if (optionCommand.options) this.applyTranslation(optionCommand.options, `${name}_${optionCommand.name}`);
+		}
+		return true;
```

Comment:
Why does this return `true` for no reason?

Reply:
lol good question